### DISCUSSION TITLE
C# - Windows x64 - Access violation on SolveWithParameters

### DISCUSSION THREAD

**Discussion by nine-2-five** [google/or-tools#3647] [Routing (and legacy CP) questions] [answered]
**What version of OR-Tools and what language are you using?**
Version: v9.5
Language: C#

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
Routing with PDP

**What operating system (Linux, Windows, ...) and version?**
Windows 11 Pro x64

**What did you do?**
This is the Git repo containing the code with this error
https://github.com/nine-2-five/or-tools-examples/blob/master/or-tools-examples/Program.cs
It happens when Main method runs and executes SolveWithParameters

**What did you expect to see**
Routing solution printed

**What did you see instead?**
![image](https://user-images.githubusercontent.com/2500267/213859696-b46944de-c3a5-4a20-9731-d77bdefcd43f.png)

Make sure you include information that can help us debug (full error message, model Proto).

**Anything else we should know about your project / environment**
Your PDP example with distance matrix did work, until I changed it to TimeMatrix.

**Comment by lperron** [comment]
Try to add a try catch around the callback code

**Reply by nine-2-five** [reply]
I tried wrapping the whole Main in try/catch but it didn't catch anything.
I also tried this, but it didn't catch anything:
```cs
int transitCallbackIndex = routing.RegisterTransitCallback((long fromIndex, long toIndex) =>
{
	// Convert from routing variable Index to
	// distance matrix NodeIndex.
	int fromNode;
	int toNode;
	try
	{
		fromNode = manager.IndexToNode(fromIndex);
		toNode = manager.IndexToNode(toIndex);
		return data.TimeMatrix[fromNode, toNode];
	}
	catch(Exception ex)
	{
		Console.WriteLine(ex.ToString());
	}
	return 123;
});
```

**Comment by lperron** [comment, marked as answer]
As written on discord, 

```
        public int[][] PickupsDeliveries = {
            new int[] { 1, 6 }, new int[] { 2, 4 },  new int[] { 5, 1 },   new int[] { 7, 3 }   
        };
```

you are reusing the same node (1). You cannot do that.
